nntrainer/src/elem_add_layer.cpp

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0-only
+ *
+ * @file	elem_add_layer.cpp
+ * @date	30 July 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Element-Wise Layer Class for Neural Network
+ *
+ */
+
+#include <elem_add_layer.h>
+#include <layer.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+int ElemAddLayer::initialize(bool last) {
+  int status = ML_ERROR_NONE;
+  if (input_dim.getDataLen() == 1) {
+    ml_logw("Warning: the length of previous layer dimension is one");
+  }
+
+  this->last_layer = last;
+
+  /** input dimension indicates the dimension for all the inputs to follow */
+  output_dim = input_dim;
+
+  return status;
+}
+
+sharedTensor ElemAddLayer::forwarding(sharedTensor in) {
+  input = in;
+  hidden = Tensor(in->batch(), output_dim.channel(), output_dim.height(),
+                  output_dim.width());
+  hidden.setZero();
+
+  for (unsigned int idx = 0; idx < num_inputs; ++idx) {
+    if (input_dim != in.get()[idx].getDim())
+      throw std::runtime_error("Error: Element wise add layer requires same "
+                               "shape from all input layers");
+    hidden.add_i(in.get()[idx]);
+  }
+
+  return MAKE_SHARED_TENSOR(hidden);
+}
+
+sharedTensor ElemAddLayer::backwarding(sharedTensor derivative, int iteration) {
+  sharedTensor ret = std::shared_ptr<Tensor>(new Tensor[num_inputs],
+                                             std::default_delete<Tensor[]>());
+  Tensor deriv = *derivative.get();
```

Comment:
IMO `get()` is not needed. 
```suggestion
  Tensor deriv = *derivative;
```

Reply:
Done.